File: frontend-angular/src/app/modules/model/friendship.ts
export class Friendship{
    friendshipId: number = 0;
    targetUserId: number = 0;
    sourceUserId: number = 0;
    status: String = "";
    
    public constructor(obj?: any) {
        if (obj) {
            this.friendshipId = obj.friendshipId;
            this.targetUserId = obj.targetUserId;
            this.sourceUserId = obj.sourceUserId;
            this.status = obj.status;
        }
    }
}

File: praksa-vanjateodorovic-backend/app/models/Friendship.scala
package models

import dtos.FriendshipDTO
import play.api.http.Status
import play.api.libs.json.{Format, Json}

case class Friendship(
                        friendshipId: Option[Long],
                        targetUserId: Long,
                        sourceUserId: Long,
                        status: String = "PENDING"
                     )

object Friendship {
  implicit val format: Format[Friendship] = Json.format[Friendship]

  implicit def friendshipDTOToFriendship(friendshipDTO: FriendshipDTO): Friendship = Friendship(friendshipDTO.friendshipId, friendshipDTO.targetUserId, friendshipDTO.sourceUserId, "PENDING")

}